src/structures/vector.ts

```diff
@@ -1,341 +1,343 @@
-        else {
-            this.x *= a;
-            this.y *= a;
-            this.z *= a;
-        }
-
-        return this;
-    }
-
-    public divide(scalar: number): Vector3 {
-        this.x /= scalar;
-        this.y /= scalar;
-        this.z /= scalar;
-
-        return this;
-    }
-
-    // ===
-    // Immutable arithmetic 
-    // ===
-    public static add(a: Vector3, b: Vector3): Vector3 {
-        return new Vector3(a.x + b.x, a.y + b.y, a.z + b.z);
-    }
-
-    public static multiply(a: Vector3, scalar: number): Vector3;
-    public static multiply(a: Vector3, b: Vector3): Vector3;
-    public static multiply(a: Vector3, b: Vector3 | number): Vector3 {
-        // console.log(a, b);
-        if(b instanceof Vector3)
-            return new Vector3(a.x * b.x, a.y * b.y, a.z * b.z);
-
-        return new Vector3(a.x * b, a.y * b, a.z * b);
-    }
-
-    public static substract(a: Vector3, b: Vector3): Vector3 {
-        return new Vector3(a.x - b.x, a.y - b.y, a.z - b.z);
-    }
-
-    public static divide(a: Vector3, scalar: number): Vector3;
-    public static divide(a: Vector3, b: Vector3): Vector3;
-    public static divide(a: Vector3, b: Vector3 | number): Vector3 {
-        if(b instanceof Vector3)
-            return new Vector3(a.x / b.x, a.y / b.y, a.z / b.z);
-
-        return new Vector3(a.x / b, a.y / b, a.z / b);
-    }
-
-    // ===
-    // Vector multiplication
-    // ===
-    public static dot(a: Vector3, b: Vector3): number {
-        return a.x * b.x + a.y * b.y + a.z * b.z;
-    }
-
-    public static cross(a: Vector3, b: Vector3): Vector3 {
-        return new Vector3(
-            a.y * b.z - a.z * b.y,
-            a.x * b.z - a.z * b.x,
-            a.x * b.y - a.y * b.x
-        );
-    }
-
-    // ===
-    // Roations
-    // ===
-    /**
-     * Returns angle between two direction vectors
-     * @param a unit vector
-     * @param b unit vector
-     * @returns angle in radians
-     */
-    public static getAngleBetween(a: Vector3, b: Vector3): number {
-        const dotProduct = Vector3.dot(a, b);
-
-        if(dotProduct >= 1)
-            return 0;
-
-        if(dotProduct <= -1)
-            return Math.PI
-
-        return Math.acos(dotProduct);
-    }
-
-    public static calculateRotationMatrix(direction: Vector3, angle: number): Matrix {
-        const cos = Math.cos(angle);
-        const sin = Math.sin(angle);
-
-        const Q = Matrix.identity(3, 3).mul(cos);
-
-        const d = direction;
-
-        const m = direction.asColumnVector().mmul(direction.asRowVector()).mul(1 - cos);
-        const n  = new Matrix([[0, d.z, -d.y], [-d.z, 0, d.x], [d.y, -d.x, 0]]).mul(sin);
-
-        for(let i=0; i < 3; i++) {
-            for(let j=0; j < 3; j++) {
-                const value = Q.get(i, j) + m.get(i, j) - n.get(i, j);
-                
-                Q.set(i, j, value);
-            }
-        }
-
-        return Q;
-    }
+import { Matrix } from 'ml-matrix';
+
+export class Vector3D {
```

Comment:
Let's return the previous name 'Vector3'. It's obvious that Vector3 is in three dimensions. And this name also used in Unity game engine.